Allow players to include specific cards

rvangrinsven opened the pull request:
Intended as a mirror to the excluded cards feature. In my group we like to play without the actual expansions while still being able to include cards from all expansions. This feature allows for that.

kberg commented:
Thanks for this! I'll get to it pretty soon.

kberg commented on src/server/GameCards.ts:
```diff
@@ -134,6 +134,7 @@ export class GameCards {
     }
 
     cards = this.filterBannedCards(cards);
+    cards = this.addCustomCards(cards, this.gameOptions.includedCards);
```
Someone who wants to use the non-ares version of a card, via this process, will still get the replaced card. I don't think that's an urgent issue, but a small and surprising bug.

kberg commented:
In general this is simple and useful.  Let's see if we can get a couple of tests out of this. I'm happy to work together on that part.

rvangrinsven commented:
Hi @kberg, I've rebased and added some (very basic :) ) tests. Do you think we could merge it in this state?

If not let me know any specific scenarios I should add tests for.

kberg commented on package.json:
```diff
@@ -19,7 +19,8 @@
     "webpack": "^5.86.0",
-    "webpack-cli": "^4.5.0"
+    "webpack-cli": "^4.5.0",
+    "zod": "^3.22.4"
```
Remove this file from the PR before we merge, please.

kberg commented on package-lock.json:
Remove this file from the PR before we merge, please.

kberg commented on src/common/game/NewGameConfig.ts:
I'm so sorry, we removed Zod, so this has to go back to how it was. :/

kberg commented on src/locales/de/promo_cards.json:
Remove this file from the PR before we merge, please.

kberg commented on src/server/Game.ts:
This file looks like it has been completely reformatted. I think you have a lot of changes to undo.

kberg commented on src/server/routes/Game.ts:
Unfortunately this file has changed will disrupts your PR.

kberg commented on tests/GameCards.spec.ts:
```diff
@@ -6,8 +6,8 @@ import {CardName} from '../src/common/cards/CardName';
 
-describe('GameCards', function() {
-  it('correctly removes projectCardsToRemove', function() {
+describe('GameCards', function () {
```
Please remove all your reformats.

kberg reviewed (commented):
I don't think your merge was quite so successful unfortunately.

rvangrinsven commented:
@kberg Sorry I see I forgot to disable my automatic formatting while doing the changes. Will fix it all now.

rvangrinsven commented:
@kberg I've fixed everything and checked it, PR should be good to go again.

kberg reviewed (approved):
This is a good change. Thank you!

Sorry again for the back and forth.
